### DISCUSSION TITLE
how to raise input validation error when there are multiple duplicate keys in request payload ?

### DISCUSSION THREAD

**Discussion by ramushetty** [pydantic/pydantic#4995] [General]
Hi, how to handle the duplicate keys in the request payload, I am trying to raise the error if there are multiple keys with the same name

{"input" : "input", "input":"input","output":"output"}

class Configuration(BaseModel):
input: FilePath = 'input'
output: FilePath = 'output'

Here is the example of my fastapi implementation

from typing import Union

from fastapi import FastAPI
from pydantic import BaseModel

class Item(BaseModel):
    name: str
    description: Union[str, None] = None
    price: float
    tax: Union[float, None] = None

app = FastAPI()

@app.post("/items/")
async def create_item(item: Item):
    return item
my request body is somthing like this with duplicate name keys

{
  "name": "name1",
  "name": "name2",
  "description": "some_des",
  "price": 0,
  "tax": 0
}
but my response is

{
  "name": "name2",
  "description": "some_des",
  "price": 0,
  "tax": 0
}
actually this leads to http parameter pollution where attacker can modify my request body.

So, I am trying to validate if there are any ways to detect the duplicate keys in request body before reaching it to pydantic or after it

if there any way without pydantic also fine

**Comment by samuelcolvin** [comment]
`{"input" : "input", "input":"input","output":"output"}` would be collapsed by python itself, or by a JSON parser, so pydantic would never see it.

**Reply by ramushetty** [reply]
thanks @samuelcolvin for inputs,  is there any way to detect the duplicate presence before it reaching the pydantic?

**Reply by samuelcolvin** [reply]
Please try it for yourself, you literally can't create a dict with duplicate keys.
